base/logging.go

```diff
@@ -284,41 +292,79 @@ func (config *LogAppenderConfig) ValidateLogAppender() error {
 // The key "bw" is interpreted as a call to LogNoColor, not a key.
 func ParseLogFlags(flags []string) {
 	logLock.Lock()
+	keyMap := make(map[string]bool)
+
 	for _, key := range flags {
+		keyMap [key] = true
+	}
+
+	ParseLogFlagsMap(keyMap)
+	logLock.Unlock()
+	Logf("Enabling logging: %s", flags)
+}
+
+// Parses a map of log keys and enabled bool, probably coming from a argv flags.
+// The key "bw" is interpreted as a call to LogNoColor, not a key.
+func ParseLogFlagsMap(flags map[string]bool) {
+	for key, val := range flags {
 		switch key {
 		case "bw":
-			LogNoColor()
+			if val {
+				LogNoColor()
+			} else {
+				LogColor()
+			}
 		case "color":
-			LogColor()
+			if val {
+				LogColor()
+			} else {
+				LogNoColor()
+			}
 		case "notime":
-			LogNoTime()
+			if val {
+				LogNoTime()
+			} else {
+				LogTime()
+			}
 		default:
-			LogKeys[key] = true
+			LogKeys[key] = val
```

Comment:
If val==false, calling delete(LogKeys, val) would make subsequent calls to LogTo() marginally more efficient.  Would need to modify everywhere you're setting to val.

Reply:
disabled keys are now removed from the keymap

Q: if a simple (no + suffix) is disabled should we attept to disable the (CHANNEL<+>) variant?

Reply:
When disabling a log key also remove logkey+ and logkey++